controllers/common_controller_utils.go

```diff
@@ -232,6 +232,38 @@ func isS3ProfileManagedPeerRef(clusterPeerRef multiclusterv1alpha1.PeerRef, mirr
 	return false
 }
 
+func updateS3ProfileFields(expected *rmn.S3StoreProfile, found *rmn.S3StoreProfile) {
+	found.S3ProfileName = expected.S3ProfileName
+	found.S3Bucket = expected.S3Bucket
+	found.S3Region = expected.S3Region
+	found.S3CompatibleEndpoint = expected.S3CompatibleEndpoint
+	found.S3SecretRef.Name = expected.S3SecretRef.Name
+}
```

Comment:
Looks correct, we update only the fields want to update so other fields should be fine.